File: surf-cloud-api/src/main/java/dev/slne/surf/cloud/api/lifecycle/SurfLifecycle.java
package dev.slne.surf.cloud.api.lifecycle;

import org.jetbrains.annotations.ApiStatus.OverrideOnly;
import org.springframework.stereotype.Component;

@OverrideOnly
@Component
public interface SurfLifecycle {

  default void onLoad() {

  }

  default void onEnable() {

  }

  default void onDisable() {

  }

  default void onReload() {
    onDisable();
    onEnable();
  }
}

File: surf-cloud-api/src/main/kotlin/dev/slne/surf/cloud/api/lifecycle/SurfLifecycle.kt
package dev.slne.surf.cloud.api.lifecycle

import org.jetbrains.annotations.ApiStatus
import org.springframework.stereotype.Component

@ApiStatus.OverrideOnly
@Component
interface SurfLifecycle {
    fun onLoad() {
    }

    fun onEnable() {
    }

    fun onDisable() {
    }

    fun onReload() {
        onDisable()
        onEnable()
    }
}
